Added http request identifier

wilzonejoe opened the pull request:
- added apigatewaysetup.py [remain redundant but will be changes to unique functions after knowing the requirements]
- modify cms functions to support new api gateway call
- added get,post,options,delete,put into setup for testing purpose
- print http method in the lambda function to test if the method is working

wilzonejoe committed:
- added apigatewaysetup.py [remain redundant but will be changes to unique functions after knowing the requirements]

- modify cms functions to support new api gateway call
- added get,post,options,delete,put into setup for testing purpose
- print http method in the lambda function to test if the method is working

chownema commented on .idea/WFP.iml:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Probably shouldn't be adding your ide project setting files to the commit lol

wilzonejoe replied:
git ignore !!

chownema commented on cms_functions.py:
```diff
@@ -470,214 +471,65 @@ def create_rest_api(self):
-            
-    def api_add_options_method(self):
+
+    def create_http_method(self, methodType , path):
```
Perhaps add py doc on the param type and expected input

chownema commented on lambda/controller.py:
```diff
@@ -32,7 +32,7 @@ def handler(event, context):
     # x = Xero(credentials)
     # print x.contacts.all()
     
-    return 'hello'
```
Yas fix my mistakes 🗡  :D

wilzonejoe replied:
YAS

chownema commented on setup.py:
```diff
@@ -11,7 +11,7 @@
 
-if len(sys.argv) not in range(2, 4):
+if len(sys.argv) not in range(1, 4):
     command = ''
```
Removed our region change? should be okay, but please comment change so if we need to deploy in different region we can

wilzonejoe replied:
i didn't remove the region change i was making the deployment easier by calling "python setup.py" instead of "python setup.py <cms-name> <region>" with a default name "testapi" or else remain the same

chownema replied:
ok

chownema commented on setup.py:
```diff
@@ -22,53 +22,59 @@
+cms.create_http_method("DELETE","root")
+cms.create_http_method("POST","other")
+cms.deploy_api()
 
```
Good

chownema commented on apigatewaysetup.py:
```diff
@@ -0,0 +1,568 @@
+import botocore
+import sys
+
```
VG

wilzonejoe replied:
sorry was compiling error from pycharm XD

chownema reviewed (commented):
Other than Gitignore needed additions and the two comments good job~

wilzonejoe committed:
- Added more comments

wilzonejoe committed:
- Fixes from yesterday commit

- Constructed a brand new handler
- some changes on the put records
- experimented in the status code change [it works!! but needed to perfect it]
